core/src/main/scala/org/apache/spark/deploy/security/HiveCredentialProvider.scala

```diff
@@ -29,11 +29,10 @@ import org.apache.hadoop.io.Text
 import org.apache.hadoop.security.{Credentials, UserGroupInformation}
 import org.apache.hadoop.security.token.Token
 
-import org.apache.spark.SparkConf
 import org.apache.spark.internal.Logging
 import org.apache.spark.util.Utils
 
-private[security] class HiveCredentialProvider extends ServiceCredentialProvider with Logging {
+private[security] class HiveCredentialProvider extends HadoopDelegationTokenProvider with Logging {
```

Comment:
> build spark without hive

It's still possible to build (as in package) a Spark distribution without Hive. This code already uses reflection, and the new dependencies should be test-scope only (aside from any issues that still need to be solved).

So you need Hive to run the tests, but you don't need Hive at runtime.